python/demos/demo_periodic_gep.py

```diff
@@ -198,82 +205,143 @@ def solve_GEP_shiftinvert(A: PETSc.Mat, B: PETSc.Mat,
-
+def assemble_and_solve(boundary_condition: List[str] = ["dirichlet","periodic"],
+                       Nev: int = 10):
+    """
+    Assemble and solve the Laplace eigenvalue problem on the unit square with
+    the prescribed boundary conditions.
 
-# Save first eigenvector
-with XDMFFile(MPI.COMM_WORLD, "results/eigenvector_0.xdmf", "w", encoding=XDMFFile.Encoding.HDF5) as xdmf:
-    xdmf.write_mesh(mesh)
-    xdmf.write_function(eigvec_r[0])
+    Parameters
+    ----------
+    boundary_condition
+        First item describes b.c. on {x=0} and {x=1}
+        Second item describes b.c. on {y=0} and {y=1}
+    Nev
+        Number of requested eigenvalues. The default is 10.
+    """
+    comm = MPI.COMM_WORLD
+    # Create mesh and finite element
+    N = 50
+    mesh = create_unit_square(MPI.COMM_WORLD, N, N)
+    V = fem.FunctionSpace(mesh, ("CG", 1))
+    fdim = mesh.topology.dim - 1
+    
+    bcs = []
+    # Dirichlet boundary condition on {y=0} and {y=1}
+    if boundary_condition[1]=="dirichlet":
+        u_bc = fem.Function(V)
+        with u_bc.vector.localForm() as u_local:
+            u_local.set(0.0)
+        def dirichletboundary(x):
+            return np.logical_or(np.isclose(x[1], 0), np.isclose(x[1], 1))
+        facets = locate_entities_boundary(mesh, fdim, dirichletboundary)
+        topological_dofs = fem.locate_dofs_topological(V, fdim, facets)
+        bc = fem.dirichletbc(u_bc, topological_dofs)
+        bcs = [bc]
+    
+    # Periodic boundary condition: {x=0} -> {x=1}
+    if boundary_condition[0]=="periodic":
+        pbc_hor_slave_tag = 2
+        pbc_hor_is_slave = lambda x: np.isclose(x[0],1) # identifier for slave boundary
+        pbc_hor_is_master = lambda x: np.isclose(x[0],0) # identifier for master boundary
+        def pbc_hor_slave_to_master_map(x):
+            out_x = np.zeros(x.shape)
+            out_x[0] = x[0] - 1
+            out_x[1] = x[1]
+            out_x[2] = x[2]
+            return out_x
+        facets = locate_entities_boundary(mesh, fdim, pbc_hor_is_slave)
+        arg_sort = np.argsort(facets)
+        pbc_hor_mt = meshtags(mesh, fdim, facets[arg_sort], np.full(len(facets), pbc_hor_slave_tag, dtype=np.int32))
+        
+    # Periodic boundary condition: {y=0} -> {y=1}
+    if boundary_condition[1]=="periodic":
+        pbc_vert_slave_tag = 3
+        pbc_vert_is_slave = lambda x: np.isclose(x[1],1)
+        def pbc_vert_slave_to_master_map(x):
+            out_x = np.zeros(x.shape)
+            out_x[0] = x[0]
+            out_x[1] = x[1] - 1
+            out_x[2] = x[2]
+            return out_x
+        facets = locate_entities_boundary(mesh, fdim, pbc_vert_is_slave)
+        arg_sort = np.argsort(facets)
+        pbc_vert_mt = meshtags(mesh, fdim, facets[arg_sort], np.full(len(facets), pbc_vert_slave_tag, dtype=np.int32))
+        
+    # Create MultiPointConstraint object
+    mpc = MultiPointConstraint(V)
+    if boundary_condition[0]=="periodic":
```

Comment:
Make sure that you run flake8 on the demo

Reply:
Sorry, I did not know about this requirement.
On fmonteghetti:master, flake8 does not highlight any issues.